projects/dex-ui/src/components/Liquidity/RemoveLiquidity.tsx

```diff
@@ -0,0 +1,320 @@
+
+  const [wellTokens, setWellTokens] = useState<Token[]>();
+  const [wellLpToken, setWellLpToken] = useState<Token | null>(null);
+  const [lpTokenAmount, setLpTokenAmount] = useState<TokenValue | undefined>();
+  const [receipt, setReceipt] = useState<ContractReceipt | null>(null);
+  const [isLoadingAllBalances, setIsLoadingAllBalances] = useState(true);
+  const [removeLiquidityMode, setRemoveLiquidityMode] = useState<REMOVE_LIQUIDITY_MODE>(REMOVE_LIQUIDITY_MODE.Balanced);
+  const [singleTokenIndex, setSingleTokenIndex] = useState<number>(0);
+  const [amounts, setAmounts] = useState<LiquidityAmounts>({});
+
+  const { isLoading: isAllTokenLoading, refetch: refetchBalances } = useAllTokensBalance();
+
+  const atLeastOneAmountNonzero = useCallback(() => Object.values(amounts).filter((amount) => amount.value.gt("0")).length > 0, [amounts]);
+
+  useEffect(() => {
+    const fetching = isAllTokenLoading;
+    fetching ? setIsLoadingAllBalances(true) : setTimeout(() => setIsLoadingAllBalances(false), 500);
+  }, [isAllTokenLoading]);
+
+  useEffect(() => {
+    if (well.tokens) {
+      const tokens: Token[] = [];
+      const initialAmounts: LiquidityAmounts = {};
+      well.tokens.forEach((token, index) => {
+        token.setMetadata({ logo: images[token.symbol] ?? images.DEFAULT });
+        tokens.push(token);
+        initialAmounts[index] = TokenValue.ZERO;
+      });
+
+      setWellTokens(tokens);
+      setAmounts(initialAmounts);
+    }
+  }, [well]);
+
+  useEffect(() => {
+    if (well.lpToken) {
+      let lpTokenWithMetadata = well.lpToken;
+      lpTokenWithMetadata.setMetadata({ logo: images[well.lpToken.symbol] ?? images.DEFAULT });
+      setLpTokenAmount(undefined);
+      setWellLpToken(lpTokenWithMetadata);
+    }
+  }, [well]);
+
+  const {
+    data: balancedQuote,
+    isLoading: loadingBalancedQuote,
+    isError: balanedQuoteError
+  } = useQuery(["wells", address, removeLiquidityMode, lpTokenAmount], async () => {
+    if (!lpTokenAmount || removeLiquidityMode !== REMOVE_LIQUIDITY_MODE.Balanced) {
+      return null;
+    }
+
+    return well.removeLiquidityQuote(lpTokenAmount);
+  });
+
+  const {
+    data: oneTokenQuote,
+    isLoading: loadingOneTokenQuote,
+    isError: oneTokenQuoteError
+  } = useQuery(["wells", address, removeLiquidityMode, lpTokenAmount, singleTokenIndex], async () => {
+    if (!lpTokenAmount || removeLiquidityMode !== REMOVE_LIQUIDITY_MODE.OneToken) {
+      return null;
+    }
+
+    return well.removeLiquidityOneTokenQuote(lpTokenAmount, wellTokens![singleTokenIndex]);
+  });
+
+  const {
+    data: customRatioQuote,
+    isLoading: loadingCustomRatioQuote,
+    isError: customRatioQuoteError
+  } = useQuery(["wells", address, removeLiquidityMode, amounts], async () => {
+    if (!atLeastOneAmountNonzero() || removeLiquidityMode !== REMOVE_LIQUIDITY_MODE.Custom) {
+      return null;
+    }
+
+    return well.removeLiquidityImbalancedQuote(Object.values(amounts));
+  });
```

Comment:
should we put these in a single hook, like 
useRemoveLiquidityQuote(quoteType, amounts[], tokenIndex?), then based on the params, return the correct useQuery result?

Also, for the useQuery cache keys:
- not sure why we need `removeLIquidityMode`
- the amounts used for quote... if we really want to cache the quote result, we should change the default ttls

Reply:
Yes they do belong in a separate hook for sure. I was treating it as a "dependency" as the docs suggested.

> if we really want to cache the quote result, we should change the default ttls
Can you elaborate on this one?